streams/src/main/java/org/apache/kafka/streams/processor/internals/TopologyMetadata.java

```diff
@@ -257,6 +260,35 @@ public void registerAndBuildNewTopology(final KafkaFutureImpl<Void> future, fina
         }
     }
 
+    /**
+     * Pauses a topology by name
+     * @param topologyName Name of the topology to pause
+     */
+    public void pauseTopology(final String topologyName) {
```

Comment:
These methods could also be unit tested really easily.

Reply:
Added tests here: https://github.com/apache/kafka/pull/12161/commits/34ec8ac54ac52196bccace8054cd7784719bf409